src/Compiler/SyntaxTree/SyntaxTree.fs

```diff
@@ -391,7 +391,7 @@ type SynTupleTypeSegment =
 [<NoEquality; NoComparison; RequireQualifiedAccess>]
 type SynType =
 
-    | LongIdent of longDotId: SynLongIdent
+    | LongIdent of longDotId: SynLongIdent * isModule: bool
```

Comment:
Is it a good place to add it to?

Adding knowledge about modules to `SynType.LongIdent` seems a bit too ad-hoc as it adds a very specific piece that is relevant to a single situation to a very commonly used type. Maybe a separate `SynType` case would work better? Similar to cases where `const` is at the front.

Reply:
@nojaf, do you reckon trivia (start position of `module`) should be added too?

Reply:
I'm with @auduchinok on this one, this new module information is added to a very common type.
Would it not make sense to consider this syntactically as an application type?

```fsharp
typeof<int list>
typeof<module Meh>
```

```fsharp
Expr
   (TypeApp
      (Ident typeof, tmp.fsx (1,6--1,7),
       [App
          (LongIdent (SynLongIdent ([list], [], [None])), None,
           [LongIdent (SynLongIdent ([int], [], [None]))], [], None,
           true, tmp.fsx (1,7--1,15))], [], Some tmp.fsx (1,15--1,16),
       tmp.fsx (1,6--1,16), tmp.fsx (1,0--1,16)), tmp.fsx (1,0--1,16));
```